packages/react-server-dom-webpack/src/__tests__/ReactFlightDOMNode-test.js

```diff
@@ -152,16 +152,19 @@ describe('ReactFlightDOMNode', () => {
   }
 
   function createDelayedStream() {
-    return new Stream.Transform({
+    let resolveDelayedStream;
+    const promise = new Promise(resolve => (resolveDelayedStream = resolve));
+    const delayedStream = new Stream.Transform({
       ...streamOptions,
       transform(chunk, encoding, callback) {
-        // Artificially delay between pushing chunks.
-        setTimeout(() => {
+        // Artificially delay pushing the chunk.
+        promise.then(() => {
```

Comment:
Is this required for this change? Looks like this is changing behavior.

Reply:
Yes, without this change, the chunks are not delayed enough to provoke the error. This also improves the usage for the test added in #34301, where we're now seeing the `Connection closed` error directly (without the fix) instead of only the `controller.enqueueModel is not a function` error that resulted from the closed connection (hence the updated comment in this test), which brings it closer to what we're creating in the Edge tests.

Reply:
Ah, I didn't scroll up enough to see this is manually resolved now instead of after a single macrotask.